cime/scripts/lib/CIME/SystemTests/mvk.py

```diff
@@ -0,0 +1,211 @@
+"""
+Multivariate test for climate reproducibility using the Kolmogrov-Smirnov (K-S)
+test and based on The CESM/E3SM model's multi-instance capability is used to
+conduct an ensemble of simulations starting from different initial conditions.
+
+This class inherits from SystemTestsCommon.
+"""
+
+import stat
+import json
+import shutil
+
+from CIME.SystemTests.system_tests_common import SystemTestsCommon
+from CIME.case.case_setup import case_setup
+from CIME.hist_utils import _get_all_hist_files, BLESS_LOG_NAME
+from CIME.utils import append_testlog, get_current_commit, get_timestamp, get_model
+from CIME.test_status import *
+
+import CIME.test_status
+
+import evv4esm
+from evv4esm.__main__ import main as evv
+
+evv_lib_dir = os.path.abspath(os.path.dirname(evv4esm.__file__))
+logger = logging.getLogger(__name__)
+
+# Build executable with multiple instances
+ninst = 20
+
+
+class MVK(SystemTestsCommon):
+
+    def __init__(self, case):
+        """
+        initialize an object interface to the MVK test
+        """
+        SystemTestsCommon.__init__(self, case)
+
+        if self._case.get_value("RESUBMIT") == 0 \
+                and self._case.get_value("GENERATE_BASELINE") is False:
+            self._case.set_value("COMPARE_BASELINE", True)
+        else:
+            self._case.set_value("COMPARE_BASELINE", False)
+
+
+    def build_phase(self, sharedlib_only=False, model_only=False):
+        # Only want this to happen once. It will impact the sharedlib build
+        # so it has to happen there.
+        if not model_only:
+            logging.warn('Starting to build multi-instance exe')
+            for comp in self._case.get_values("COMP_CLASSES"):
+                self._case.set_value('NTHRDS_{}'.format(comp), 1)
+
+                ntasks = self._case.get_value("NTASKS_{}".format(comp))
+
+                self._case.set_value('NTASKS_{}'.format(comp), ntasks*ninst)
+                if comp != 'CPL':
+                    self._case.set_value('NINST_{}'.format(comp), ninst)
+
+            self._case.set_value('ATM_NCPL', 18)
+
+            self._case.flush()
+
+            case_setup(self._case, test_mode=False, reset=True)
+
+        self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
+
+        # =================================================================
+        # Run-time settings.
+        # Do this already in build_phase so that we can check the xml and
+        # namelist files before job starts running.
+        # =================================================================
+
+        # namelist specifications for each instance
+        for iinst in range(1, ninst+1):
+            with open('user_nl_cam_{:04d}'.format(iinst), 'w') as nl_atm_file:
+                nl_atm_file.write('new_random = .true.\n')
+                nl_atm_file.write('pertlim = 1.0e-10\n')
+                nl_atm_file.write('seed_custom = {}\n'.format(iinst))
+
+
+    def run_phase(self):
```

Comment:
This override is unnecessary since the implementation is an exact match of the parent's.

Reply:
Ah right, yep! I forgot to clean that part up; previous iterations did stuff there. Thanks!